app/models/highlight.rb

```diff
@@ -19,6 +19,14 @@ class Highlight < ApplicationRecord
 
   before_validation :normalize_color, :normalize_source_ids
 
+  scope :by_source_type, ->(source_type) { where(source_type: source_type) }
+  scope :by_color, ->(color) { where(color: color) }
+  scope :by_user, ->(user_id) { where(user_uuid: user_id) }
+  scope :by_order, ->(order_direction) { order(created_at: order_direction.to_sym) }
```

Comment:
If this is pagination order it should be based on the two highlight order fields.